### DISCUSSION TITLE
Understand base64 encode id for Github v4 API

### DISCUSSION THREAD

**Discussion by shivam05011996** [community/community#24858] [Apps, API and Webhooks]
<p>Recently when querying to github, the ID field for an entity decribes it self as “<em>Represents a unique identifier that is Base64 obfuscated. It is often used to refetch an object or as key for a cache. The ID type appears in a JSON response as a String; however, it is not intended to be human-readable. When expected as an input type, any string (such as “VXNlci0xMA==”) or integer (such as 4</em>_) input value will be accepted as an ID_.”</p>
<p>When I decode an Id as -</p>
<pre><code>&gt;&gt;&gt; b64decode('MDQ6VXNlcjMzODI4OTQy').decode('utf-8')
'04:User33828942'
</code></pre>
<p>“User” signifies entity, and the number signifies databaseID?</p>
<p>What does “04” prefixed to “:” mean/ signify?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>The underlying structure of the IDs is undocumented and subject to change at any time.</p>
<p>I hope that helps.</p>

**Comment by carlaarchuleta707-lgtm** [comment]
`>>>` b64decode('MDQ6VXNlcjMzODI4OTQy').decode('utf-8')
'04:User33828942'
